src/tools/rados/rados.cc

```diff
@@ -3342,18 +3342,25 @@ static int rados_tool_common(const std::map < std::string, std::string > &opts,
       return 1;
     }
 
-    set<string> out_keys;
-    ret = io_ctx.omap_get_keys(nargs[1], "", LONG_MAX, &out_keys);
-    if (ret < 0) {
-      cerr << "error getting omap key set " << pool_name << "/"
-	   << nargs[1] << ": "  << cpp_strerror(ret) << std::endl;
-      return 1;
-    }
+    string last_read = "";
+    int MAX_READ = 512;
+    do {
+      set<string> out_keys;
+      ret = io_ctx.omap_get_keys(nargs[1], last_read, MAX_READ, &out_keys);
+      if (ret < 0) {
+        cerr << "error getting omap key set " << pool_name << "/"
+             << nargs[1] << ": "  << cpp_strerror(ret) << std::endl;
+        return 1;
+      }
+      ret = out_keys.size();
 
-    for (set<string>::iterator iter = out_keys.begin();
-	 iter != out_keys.end(); ++iter) {
-      cout << *iter << std::endl;
-    }
+      for (set<string>::iterator iter = out_keys.begin();
+           iter != out_keys.end(); ++iter) {
+        last_read = *iter;
```

Comment:
please consider transpose these two statements, so we can `std::move()` from the element.

Reply:
good